MM-53813: Fix(accessibility): on channels page

saturninoabril opened the pull request:
#### Summary
- Fixed issues found after running [FastPass](https://accessibilityinsights.io/docs/web/getstarted/fastpass/) in the Channels page.
- Added Playwright accessibility test

Notes:
- Requires PR from dynamic-virtualized-list: https://github.com/mattermost/dynamic-virtualized-list/pull/28
- `color-contrast` ratio requirement to be addressed separately - https://mattermost.atlassian.net/browse/MM-53813

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53813


#### Screenshots
Before:
<img width="1261" alt="Screenshot 2023-07-26 at 4 31 50 PM" src="https://github.com/mattermost/mattermost/assets/5334504/b1f75d98-5011-43e7-9e74-633e01741efe">

Playwright test:
<img width="300" alt="Screenshot 2023-07-26 at 7 06 20 AM" src="https://github.com/mattermost/mattermost/assets/5334504/57290e91-ba35-4cf8-9bf0-6cd17b9e5346">


#### Release Note

```release-note
NONE
```

M-ZubairAhmed commented:
You are on 🔥 with these a11y fixes. Awesome

saturninoabril commented on webapp/channels/package.json:
```diff
@@ -32,7 +32,7 @@
     "css-vars-ponyfill": "2.4.8",
     "date-fns": "2.29.3",
-    "dynamic-virtualized-list": "github:mattermost/dynamic-virtualized-list#1c330717d6778315a40e70137ace38247c6a1d0f",
+    "dynamic-virtualized-list": "github:mattermost/dynamic-virtualized-list#9695a52781981db77333607343ea62eac828bda4",
```
Will update once the related PR is merged.

yasserfaraazkhan reviewed (approved):
thank you for the changes @saturninoabril .

just a general thought 
I was wondering if there's away we can also capture in pr the before/ after or convert that PR meets A,AA level of success criteria through applied fixes

saturninoabril commented on webapp/channels/src/components/__snapshots__/searchable_channel_list.test.tsx.snap:
```diff
@@ -65,7 +65,7 @@ exports[`components/SearchableChannelList should match init snapshot 1`] = `
   <div
     className="more-modal__list"
-    role="application"
+    role="search"
```
This fixes the issue in the screenshot.
<img width="537" alt="Screenshot 2023-07-26 at 4 34 28 PM" src="https://github.com/mattermost/mattermost/assets/5334504/649e185c-51f6-4cc2-b677-69bd494a4362">

saturninoabril commented on webapp/channels/src/components/channel_header/__snapshots__/channel_header.test.tsx.snap:
```diff
@@ -4700,7 +4685,6 @@ exports[`components/ChannelHeader should render the pinned icon with the pinned
                     </span>
                   </strong>
                   <span
-                    aria-label="dropdown icon"
```
This fixes the issue in the screenshot. Such ARIA label is redundant to the parent (button) ARIA label as `channel menu` which acts as single interactive element. Dropdown icon is not accessible/interactive by its own alone. 
<img width="588" alt="Screenshot 2023-07-26 at 4 45 44 PM" src="https://github.com/mattermost/mattermost/assets/5334504/72d37623-f6d5-4709-98ca-c6a73f58fa5b">

saturninoabril commented on webapp/channels/src/components/channel_header/channel_header.tsx:
```diff
@@ -785,7 +785,6 @@ class ChannelHeader extends React.PureComponent<Props, State> {
                             <span
                                 id='channelHeaderDropdownIcon'
                                 className='icon icon-chevron-down header-dropdown-chevron-icon'
-                                aria-label={formatMessage({id: 'generic_icons.dropdown', defaultMessage: 'Dropdown Icon'}).toLowerCase()}
```
This fixes the issue in the screenshot. Such ARIA label is redundant to the parent (button) ARIA label as `channel menu` which acts as single interactive element. Dropdown icon is not accessible/interactive by its own alone. 
<img width="588" alt="Screenshot 2023-07-26 at 4 45 44 PM" src="https://github.com/mattermost/mattermost/assets/5334504/72d37623-f6d5-4709-98ca-c6a73f58fa5b">

saturninoabril commented on webapp/channels/src/components/post/post_component.tsx:
```diff
@@ -513,7 +513,6 @@ const PostComponent = (props: Props): JSX.Element => {
             {(isSearchResultItem || (props.location !== Locations.CENTER && (props.isPinnedPosts || props.isFlaggedPosts))) && <DateSeparator date={currentPostDay}/>}
             <PostAriaLabelDiv
                 ref={postRef}
-                role='listitem'
```
This and the changes at [post_list_virtualized/post_list_virtualized.tsx](https://github.com/mattermost/mattermost/pull/24122/files#diff-61c4a17151b88abdf551ba2bc3a2298363473a02fd9352a8b878002fcd831069), and on [dynamic-virtualized-list](https://github.com/mattermost/dynamic-virtualized-list/pull/28) fixed the issue in the screenshot. It properly set the ARIA role's parent-children hierarchy in the post list content.
<img width="884" alt="Screenshot 2023-07-26 at 4 52 14 PM" src="https://github.com/mattermost/mattermost/assets/5334504/4ae993a2-a527-493e-a041-c7c6ae88aa08">

saturninoabril commented on webapp/channels/src/components/post_priority/post_priority_picker_overlay.tsx:
```diff
@@ -104,6 +101,7 @@ function PostPriorityPickerOverlay({
                     className={classNames({control: true, active: pickerOpen})}
                     disabled={disabled}
                     type='button'
+                    aria-label={messagePriority}
```
This fixes the issue in the screenshot.
<img width="544" alt="Screenshot 2023-07-26 at 4 57 59 PM" src="https://github.com/mattermost/mattermost/assets/5334504/88fc7d8f-f6f5-4758-9345-9ac86aa2bc97">

saturninoabril commented on webapp/channels/src/components/sidebar/__snapshots__/invite_members_button.test.tsx.snap:
```diff
@@ -68,17 +68,17 @@ exports[`components/sidebar/invite_members_button should match snapshot 1`] = `
-              aria-label="Invite Users dialog"
+              aria-label="Invite Members dialog"
               className="style--none intro-links color--link cursor--pointer"
-              id="introTextInvite"
```
This fixes the issue of duplicate ID.
<img width="847" alt="Screenshot 2023-07-26 at 5 02 27 PM" src="https://github.com/mattermost/mattermost/assets/5334504/e3cc5ed7-547f-4d1d-b7af-a8ca5fe3f41c">

saturninoabril commented on webapp/channels/src/components/sidebar/__snapshots__/invite_members_button.test.tsx.snap:
```diff
@@ -93,7 +93,7 @@ exports[`components/sidebar/invite_members_button should match snapshot 1`] = `
                     Invite Members
                   </span>
                 </FormattedMessage>
-              </li>
```
This and the next one for "Add Channels" fixes the issue in the screenshot.

saturninoabril commented:
@yasserfaraazkhan Was pulled out on other things but that's my plan too.
- added before fix in the description; and
- added comments to each changes and issues being addressed.

saturninoabril commented:
@amyblais I'm adding this to 8.1 milestone to address customer issues on Accessibility.

saturninoabril commented:
/update-branch

saturninoabril commented on webapp/channels/src/components/sidebar/sidebar_category_header.tsx:
```diff
@@ -40,6 +40,14 @@ type Props = StaticProps & {
+    // by screen readers or can cause focus problems for assistive technologies.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    delete dragHandleProps?.role;
```
This fixes the issue in the screenshot.
<img width="834" alt="Screenshot 2023-07-26 at 5 52 28 PM" src="https://github.com/mattermost/mattermost/assets/5334504/0af7761a-0b9a-4ee4-90ad-42d55a1e3c1c">

M-ZubairAhmed commented on webapp/channels/src/components/sidebar/sidebar_category_header.tsx:
```diff
@@ -40,6 +40,14 @@ type Props = StaticProps & {
+    // by screen readers or can cause focus problems for assistive technologies.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    delete dragHandleProps?.role;
```
nit: to remove eslint supressor

```
if (dragHandleProps && dragHandleProps.role) {
        Reflect.deleteProperty(dragHandleProps, 'role');
    }
    
```

saturninoabril replied:
Thanks Zubair, applied and done.

saturninoabril commented:
/update-branch

saturninoabril commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/357077) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/357077)

saturninoabril commented:
@asaadmahmood Once all good in E2E,I'm merging this so it can get into 8.1 but let me know if you have any comment.

saturninoabril commented:
/e2e-test

saturninoabril commented:
/e2e-test

mattermost-build commented:
A pipeline with the same environment variables is already running. 
 Please cancel it first with /e2e-cancel, or specify different environment variables.

saturninoabril commented:
/e2e-cancel

mattermost-build commented:
Looks like nothing had to be canceled.

saturninoabril commented:
/e2e-test

mattermost-build commented:
A pipeline with the same environment variables is already running. 
 Please cancel it first with /e2e-cancel, or specify different environment variables.

mattermost-build commented:
Cherry pick is scheduled.

mattermost-build commented:
Error trying doing the automated Cherry picking. Please do this manually

```
+++ Updating remotes...
Fetching upstream
Failed to add the RSA host key for IP address '140.82.114.4' to the list of known hosts (/app/.ssh/known_hosts).
From github.com:mattermost/mattermost
   0489526377..10c30428f0  MM-37984              -> upstream/MM-37984
   1c21f59fa5..28542eed51  MM-53609-remove-channels-from-archived-teams-in-search -> upstream/MM-53609-remove-channels-from-archived-teams-in-search
   d7697f3b39..7ad67f0b7a  cloud                 -> upstream/cloud
   62104e67fa..73dd6b9192  fix-e2e-macos-support -> upstream/fix-e2e-macos-support
 * [new branch]            fix-e2e-win-support   -> upstream/fix-e2e-win-support
   0b55a40b21..3322b82026  gm_to_channel         -> upstream/gm_to_channel
   a47af39d80..93a2c3281a  master                -> upstream/master
   ac1a52ead6..417b0dab81  release-7.10          -> upstream/release-7.10
   1ddb83a765..522755acef  release-7.8           -> upstream/release-7.8
   9832155043..1e44bbfefe  release-8.0           -> upstream/release-8.0
   a1855ae665..326cffe0d8  release-8.1           -> upstream/release-8.1
Fetching origin
Failed to add the RSA host key for IP address '140.82.113.3' to the list of known hosts (/app/.ssh/known_hosts).
From github.com:mattermost-build/mattermost
   781498f7f8..286b986561  automated-cherry-pick-of-mattermost-#24143-upstream-cloud -> origin/automated-cherry-pick-of-mattermost-#24143-upstream-cloud
+++ Updating remotes done...
+++ Creating local branch automated-cherry-pick-of-mattermost-#24122-upstream-release-8.1-1690557438
Switched to a new branch 'automated-cherry-pick-of-mattermost-#24122-upstream-release-8.1-1690557438'
Branch 'automated-cherry-pick-of-mattermost-#24122-upstream-release-8.1-1690557438' set up to track remote branch 'release-8.1' from 'upstream'.

+++ About to attempt cherry pick of PR #24122 with merge commit 93a2c3281ad896a84fc2a3fd6a52708a765a7dd9.

error: could not apply 93a2c3281a... MM-53813: Fix(accessibility): on channels page (#24122)
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'

+++ Conflicts detected:

UU webapp/channels/src/components/post_edit_history/__snapshots__/post_edit_history.test.tsx.snap
Aborting.

+++ Aborting in-progress git cherry-pick.

+++ Returning you to the master branch and cleaning up.

```

amyblais commented:
/cherry-pick cloud

mattermost-build commented:
Cherry pick is scheduled.
